examples/gno.land/r/gnoland/faucet/faucet.gno

```diff
@@ -4,65 +4,95 @@ import (
 	"errors"
 	"std"
 
+	"gno.land/p/demo/avl"
 	"gno.land/p/demo/ufmt"
 )
 
 var (
 	// configurable by admin.
-	gAdminAddr       std.Address = "g1u7y667z64x2h7vc6fmpcprgey4ck233jaww9zq"
-	gControllerAddr  std.Address = "g1u7y667z64x2h7vc6fmpcprgey4ck233jaww9zq"
-	gPerTransferSend             = std.Coins{std.Coin{"ugnot", 1000000}}
-	gInPause                     = false
-	gMessage                     = "# Community Faucet.\n\n"
+	gAdminAddr std.Address = std.Address("g1jg8mtutu9khhfwc4nxmuhcpftf0pajdhfvsqf5")
+
+	gControllers     = avl.NewMutTree()
+	gControllersSize = 10 // limit it to 10
+
+	gInPause = false
+
+	gMessage = "# Community Faucet.\n\n"
 
 	// internal vars, for stats.
 	gTotalTransferred std.Coins
 	gTotalTransfers   = uint(0)
+
+	// per request limit, 350 gnot
+	gLimit std.Coin = std.Coin{"ugnot", 350000000}
 )
 
-func Transfer(to std.Address) error {
+func Transfer(to std.Address, send int64) string {
 	if err := assertIsController(); err != nil {
-		return err
+		return err.Error()
 	}
 
 	if gInPause {
-		return errors.New("faucet in pause")
+		return errors.New("faucet in pause").Error()
 	}
 
-	send := gPerTransferSend
+	// limit the per request
+	if send > gLimit.Amount {
+		return errors.New("Per request limit " + gLimit.String() + " exceed").Error()
+	}
+	sendCoins := std.Coins{std.Coin{Denom: "ugnot", Amount: send}}
 
-	gTotalTransferred = gTotalTransferred.Add(send)
+	gTotalTransferred = gTotalTransferred.Add(sendCoins)
 	gTotalTransfers++
 
-	banker := std.GetBanker(std.BankerTypeOrigSend)
+	banker := std.GetBanker(std.BankerTypeRealmSend)
 	pkgaddr := std.GetOrigPkgAddr()
-	banker.SendCoins(pkgaddr, to, send)
-	return nil
+	banker.SendCoins(pkgaddr, to, sendCoins)
+	return ""
+}
+
+func GetPerTransferLimit() int64 {
+	return gLimit.Amount
 }
 
 func Render(path string) string {
-	banker := std.GetBanker(std.BankerTypeOrigSend)
+	banker := std.GetBanker(std.BankerTypeRealmSend)
 	balance := banker.GetCoins(std.GetOrigPkgAddr())
 
-	output := gMessage
+	output := path + gMessage
 	if gInPause {
 		output += "Status: inactive.\n"
 	} else {
 		output += "Status: active.\n"
 	}
 	output += ufmt.Sprintf("Balance: %s.\n", balance.String())
-	output += ufmt.Sprintf("Rewarded: %s (in %d times).\n", gTotalTransferred.String(), gTotalTransfers)
+	output += ufmt.Sprintf("Total transfers: %s (in %d times).\n\n", gTotalTransferred.String(), gTotalTransfers)
+
+	output += "Package address: " + std.GetOrigPkgAddr().String() + "\n\n"
+	output += ufmt.Sprintf("Admin: %s\n\n ", gAdminAddr.String())
+	output += ufmt.Sprintf("Controllers:\n\n ")
+
+	for i := 0; i < gControllersSize; i++ {
+
```

Comment:
remove newline

Reply:
I ran gofmt, gofumpt.  these are valid new lines.  These new line act as a logical grouping and are easy to read.

Reply:
I also prefer without new lines. And yes, this is valid go.

I think we should discuss and choose styling guidelines, then try to configure a linter. I will open an issue.

Reply:
-> https://github.com/gnolang/gno/issues/453